.gitlab-ci.yml

```diff
@@ -763,6 +763,26 @@ workflow:
 
 # New E2E related rules
 
+.on_e2e_or_fakeintake_changes_or_manual:
+  - <<: *if_main_branch
+  - !reference [.except_mergequeue]
+  - changes:
+      paths:
+        - test/new-e2e/fakeintake/**/*
+      compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
+    variables:
+      FAKEINTAKE_IMAGE_OVERRIDE: "public.ecr.aws/datadog/fakeintake:v$CI_COMMIT_SHORT_SHA"
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Would move this to https://github.com/DataDog/datadog-agent/blob/main/.gitlab/dev_container_deploy/fakeintake.yml#L15 to keep publishing and target image in the same context

Reply:
We used to define all the gitlab rules in `.gitlab-ci.yaml` I think that's better for readability. If I move it I cannot reuse the yaml `*fakeintake_changes` anchor, which would duplicate some codes